syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -0,0 +1,198 @@
+        return result
+
+    __add__ = add
+
+    def __iadd__(self, other):
+        """Add two fixed precision tensors together.
+        """
+        self.child = self.add(other).child
+
+        return self
+
+    def fix_large_precision(self):
+        self.child = self._create_internal_tensor()
+        return self
+
+    def float_precision(self):
+        """
+        Restore the tensor expressed now as a matrix for each original item.
+
+        Returns:
+            tensor: the original tensor.
+        """
+        # We need to pass the PyTorch tensor to Numpy to allow the intermediate large number.
+        # An alternative would be to iterate through the PyTorch tensor and apply the restore function.
+        # This however wouldn't save us from creating a new tensor
+        ndarray = self.child.numpy()
+        result = self._restore_tensor_into_numbers(ndarray, internal_precision[self.internal_type])
+        return torch.from_numpy(result.reshape(ndarray.shape[:-1]).astype(np.float32))
+        # At this point the value is an object type. Force cast to float before creating torch.tensor
+
+    def _restore_tensor_into_numbers(self, number_array, precision):
+        """Creates an numpy array containing the original numbers"""
+        number_array = number_array.reshape(-1, number_array.shape[-1])
+        result = []
+        for elem in number_array:
+            result.append(self._restore_number(elem, precision))
+        return np.array(result)
+
+    @staticmethod
+    def _adjust_to_shape(to_adjust, shape, fill_value=0) -> torch.Tensor:
+        """Numbers with the same precision can be represented with different matrices.
+        This function adjusts the shapes of two tensors to be the same and fill the empty cells with the
+        provided `fill_value`.
+        We assume only the last dimension needs to be adjusted.
+        Original input tensors should have the same dimensions.
+        """
+        diff_dim = shape[-1] - to_adjust.shape[-1]
+        new_shape = to_adjust.shape[:-1] + (diff_dim,)
+
+        if not fill_value:
+            filler = torch.zeros(size=new_shape, dtype=to_adjust.dtype)
+        else:
+            filler = torch.ones(size=new_shape, dtype=to_adjust.dtype)
+        return torch.cat([filler, to_adjust], len(shape) - 1)
+
+    def _split_number(self, number, bits):
+        """Splits a number in numbers of a smaller power.
+
+        Args:
+            number (int): the number to split.
+            bits (int): the bits to use in the split.
+
+        Returns:
+            list: a list of numbers representing the original one.
+
+        """
+        if not number:
+            return [number]
+
+        if number > 0:
+            sign = 1
+        else:
+            sign = -1
+            number = number * (-1)
+
+        base = 2 ** bits
+        number_parts = []
+        while number:
+            number, part = divmod(number, base)
+            number_parts.append(part * sign)
```

Comment:
Maybe we need a check to be sure that we do not try to put a negative number in a LPT made of uint